File: finanze/domain/forecast.py
from datetime import date
from typing import Optional
from uuid import UUID

from domain.dezimal import Dezimal
from domain.global_position import EntitiesPosition
from pydantic.dataclasses import dataclass


@dataclass
class ForecastRequest:
    target_date: date
    entities: Optional[list[UUID]] = None
    excluded_entities: Optional[list[UUID]] = None
    avg_annual_market_increase: Optional[Dezimal] = None
    avg_annual_crypto_increase: Optional[Dezimal] = None
    avg_annual_commodity_increase: Optional[Dezimal] = None


@dataclass
class CashDelta:
    currency: str
    amount: Dezimal


@dataclass
class RealEstateEquityForecast:
    id: UUID
    equity_now: Optional[Dezimal]
    equity_at_target: Optional[Dezimal]
    principal_outstanding_now: Optional[Dezimal]
    principal_outstanding_at_target: Optional[Dezimal]
    currency: str


@dataclass
class ForecastResult:
    target_date: date
    positions: EntitiesPosition
    cash_delta: list[CashDelta]
    real_estate: list[RealEstateEquityForecast]
    crypto_appreciation: Dezimal
    commodity_appreciation: Dezimal

File: mobile/src/domain/core/forecast.ts
import { EntitiesPosition } from "./globalPosition"
import { Dezimal } from "@/domain/dezimal"

export interface ForecastRequest {
  targetDate: string
  avgAnnualMarketIncrease?: Dezimal | null
  avgAnnualCryptoIncrease?: Dezimal | null
  avgAnnualCommodityIncrease?: Dezimal | null
}

export interface CashDelta {
  currency: string
  amount: Dezimal
}

export interface RealEstateEquityForecast {
  id: string
  equityNow: Dezimal | null
  equityAtTarget: Dezimal | null
  principalOutstandingNow: Dezimal | null
  principalOutstandingAtTarget: Dezimal | null
  currency: string
}

export interface ForecastResult {
  targetDate: string
  positions: EntitiesPosition
  cashDelta: CashDelta[]
  realEstate: RealEstateEquityForecast[]
  cryptoAppreciation: Dezimal
  commodityAppreciation: Dezimal
}
